src/test/java/com/google/api/codegen/testdata/java_mock_service_impl_library.baseline

```diff
@@ -297,8 +284,7 @@ public class MockLibraryServiceImpl implements LibraryService  {
   public void updateBookIndex(UpdateBookIndexRequest request,
       StreamObserver<Empty> responseObserver) {
     if (responses != null && !responses.isEmpty()) {
-      Empty response = (Empty) responses.get(0);
-      responses.remove(0);
+      Empty response = (Empty) responses.poll();
```

Comment:
Hmmm ok I see what you mean. It is hard to predict if the responses queue would be empty due to an user error but in my opinion it maybe relatively rare. It would be more common to be caused by a bug in the generator. With that said, we probably should directly throw an exception immediately in this case.